care/facility/api/viewsets/patient.py

```diff
@@ -633,19 +636,68 @@ def transfer(self, request, *args, **kwargs):
         return Response(data=response_serializer.data, status=status.HTTP_200_OK)
 
 
+class DischargePatientFilterSet(PatientFilterSet):
+    last_consultation_field = "last_discharge_consultation"
+
+    def filter_by_bed_type(self, queryset, name, value):
+        if not value:
+            return queryset
+
+        values = value.split(",")
+        filter_q = Q()
+        last_consultation_bed_ids = (
+            ConsultationBed.objects.filter(end_date__isnull=False)
+            .order_by("consultation__patient_id", "-end_date")
+            .distinct("consultation__patient_id")
+        )
+
+        if "None" in values:
+            filter_q |= ~Q(
+                last_discharge_consultation__id__in=Subquery(
+                    last_consultation_bed_ids.values_list("consultation_id", flat=True)
+                )
+            )
+            values.remove("None")
+
+        if isinstance(values, list) and len(values) > 0:
+            filter_q |= Q(
+                last_discharge_consultation__id__in=Subquery(
+                    ConsultationBed.objects.filter(
+                        id__in=Subquery(
+                            last_consultation_bed_ids.values_list("id", flat=True)
+                        ),
+                        bed__bed_type__in=values,
+                    ).values_list("consultation_id", flat=True)
+                )
+            )
+
+        return queryset.filter(filter_q)
+
+
 @extend_schema_view(tags=["patient"])
 class FacilityDischargedPatientViewSet(GenericViewSet, mixins.ListModelMixin):
     permission_classes = (IsAuthenticated, DRYPermissions)
     lookup_field = "external_id"
     serializer_class = PatientListSerializer
     filter_backends = (
+        PatientDRYFilter,
         filters.DjangoFilterBackend,
         rest_framework_filters.OrderingFilter,
         PatientCustomOrderingFilter,
     )
-    filterset_class = PatientFilterSet
+    filterset_class = DischargePatientFilterSet
     queryset = (
```

Comment:
@shivankacker This seems like a pretty complex query. Can you add a simple comment about what this filter does?

Reply:
@anroopak I have added some documentation for the filter, please have a look